From 2e50e9ee1806bb019c3cdc79bfe594a923fa483d Mon Sep 17 00:00:00 2001
From: Claudiu Popa <pcmanticore@gmail.com>
Date: Sat, 8 Jul 2017 11:46:39 +0300
Subject: [PATCH 1/2] Remove module that wasn't actually moved. Close #1565

---
 pylint/checkers/python3.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pylint/checkers/python3.py b/pylint/checkers/python3.py
index f70aa28..f38e6f4 100644
--- a/pylint/checkers/python3.py
+++ b/pylint/checkers/python3.py
@@ -500,7 +500,7 @@ class Python3Checker(checkers.BaseChecker):
             'dircache', 'dl', 'exception', 'fpformat', 'htmllib', 'ihooks', 'imageop', 'imputil',
             'linuxaudiodev', 'md5', 'mhlib', 'mimetools', 'MimeWriter', 'mimify', 'multifile',
             'mutex', 'new', 'popen2', 'posixfile', 'pure', 'rexec', 'rfc822', 'sha', 'sgmllib',
-            'sre', 'stat', 'stringold', 'sunaudio', 'sv', 'test.testall', 'thread', 'timing',
+            'sre', 'stringold', 'sunaudio', 'sv', 'test.testall', 'thread', 'timing',
             'toaiff', 'user', 'urllib2', 'urlparse'
         ]),
         'deprecated-string-function': {
-- 
2.9.5

